### PR TITLE
[#24] UI implementation to submit survey

### LINKED ISSUE #24: [UI] As a user, I can submit my survey response
## Why
At the final step, users must be able to submit their responses. The mobile app must show the Submit button instead of the Next button to inform users that the app will submit their response on the next action.

## Acceptance Criteria
- When a user reaches the last question, (The current question number equals to the number of questions)
- Hide the Next button
- Display the Submit button

Image

## Design
[Take Survey - Last Question](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?page-id=0%3A1&node-id=34%3A1172&scaling=min-zoom)

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #24 

## What happened 👀

At the final step, users must be able to submit their responses. The mobile app must show the Submit button instead of the Next button to inform users that the app will submit their response on the next action.

- When a user reaches the last question, (The current question number equals to the number of questions)
- Hide the Next button
- Display the Submit button

## Insight 📝

- Reuse start survey button to show `Submit` button
- Refactor functions to be reusable

## Proof Of Work 📹

https://user-images.githubusercontent.com/32578035/188577419-da117e7d-70c3-4174-8320-85664cd41d59.mp4

### CODE REVIEW

**Review by AVI5HEK — APPROVED** [review]
Sweet 🍭

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyDetailScreen.kt [hunk 1]
```diff
@@ -147,7 +154,7 @@ fun SurveyDetailScreen(
                 .semantics { contentDescription = startSurveyDescription }
         )
         NextQuestionButton(
-            showButton = currentPage > 0,
+            showButton = currentPage > 0 && !isLastPage,
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyDetailScreen.kt]: How about reusing `isStartPage` too? 😄 
```suggestion
            showButton = !isStartPage && !isLastPage,
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyDetailScreen.kt [hunk 2]
```diff
@@ -197,13 +205,13 @@ fun StartSurveyButton(
     ) {
         if (it) {
             Button(
-                onClick = { onNextSlide() },
+                onClick = { onClick() },
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyDetailScreen.kt:208]: Nice job making it more generic 😉

**Review by ryan-conway — COMMENTED** [review]
One comment, the rest LGTM 🙏🏻

**Review by ryan-conway — APPROVED** [review]
LGTM 💪🏻

**Pull request merged**